finality-aleph/src/data_io/data_store.rs

```diff
@@ -0,0 +1,636 @@
+        self.num_triggers_registered_since_last_pruning += 1;
+        if number > self.highest_finalized_num {
+            self.event_triggers
+                .entry(ChainEvent::Finalized(number))
+                .or_insert_with(HashSet::new)
+                .insert(proposal.clone());
+        }
+    }
+
+    fn register_next_finality_trigger(&mut self, proposal: &AlephProposal<B>) {
+        if self.highest_finalized_num < proposal.number_below_branch() {
+            self.register_finality_trigger(proposal, proposal.number_below_branch());
+        } else if self.highest_finalized_num < proposal.number_top_block() {
+            self.register_finality_trigger(
+                proposal,
+                self.highest_finalized_num + NumberFor::<B>::one(),
+            );
+        }
+    }
+
+    fn on_block_finalized(&mut self, block: BlockHashNum<B>) {
+        if self.highest_finalized_num < block.num {
+            // We don't assume block.num = self.highest_finalized_num + 1 as the finality import queue does
+            // not quite guarantee this.
+            self.proposal_status_provider
+                .lock()
+                .on_block_finalized(block.clone());
+            let old_num = self.highest_finalized_num;
+            let new_num = block.num;
+            self.highest_finalized_num = new_num;
+            // We activate all finality triggers in [old_num + 1, block.num].
+            let mut num: NumberFor<B> = old_num + NumberFor::<B>::one();
+            while num <= new_num {
+                if let Some(proposals_to_bump) =
+                    self.event_triggers.remove(&ChainEvent::Finalized(num))
+                {
+                    for proposal in proposals_to_bump {
+                        self.bump_proposal(&proposal);
+                    }
+                }
+                num += NumberFor::<B>::one();
+            }
+        }
+    }
+
+    fn on_block_imported(&mut self, header: B::Header) {
+        let block: BlockHashNum<B> = (header.hash(), *header.number()).into();
+        self.proposal_status_provider
+            .lock()
+            .on_block_imported(header);
+        if let Some(proposals_to_bump) = self.event_triggers.remove(&ChainEvent::Imported(block)) {
+            for proposal in proposals_to_bump {
+                self.bump_proposal(&proposal);
+            }
+        }
+    }
+
+    fn on_proposal_available(&mut self, proposal: &AlephProposal<B>) {
+        if let Some(proposal_info) = self.pending_proposals.remove(proposal) {
+            for id in proposal_info.messages {
+                self.remove_proposal_from_pending_message(proposal, id);
+            }
+        }
+    }
+
+    // Makes an availability check for `data` and updates its status. Outputs whether the bump resulted in
+    // this proposal becoming available.
+    fn bump_proposal(&mut self, proposal: &AlephProposal<B>) -> bool {
```

Comment:
This has a name that reflects what this method does, which I think is more valuable than a name which returns what it outputs, because the outputs is of minor importance here.

Reply:
It makes the only place where the return value is use really hard to read. Please add a comment there (line 320) if you want to keep this name.

Reply:
Ok, added a comment.